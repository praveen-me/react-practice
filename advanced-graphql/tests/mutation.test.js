const gql = require("graphql-tag");
const createTestServer = require("./helper");
const CREATE_POST = gql`
  mutation($input: NewPostInput!) {
    createPost(input: $input) {
      message
    }
  }
`;

describe("mutation", () => {
  const message = { message: "Hello World Again" };

  test("create post", async () => {
    const { mutate } = createTestServer({
      user: { id: 1 },
      models: {
        Post: {
          createOne() {
            return {
              ...message
            };
          }
        },
        user: { id: 1 }
      }
    });

    const res = await mutate({
      query: CREATE_POST,
      variables: {
        input: {
          ...message
        }
      }
    });
    expect(res).toMatchSnapshot();
  });
});
