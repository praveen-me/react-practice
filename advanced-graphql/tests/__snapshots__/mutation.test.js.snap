// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutation create post 1`] = `
Object {
  "data": Object {
    "createPost": Object {
      "message": "Hello World Again",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
